sql/updates/latest-dev.sql

```diff
@@ -17,3 +17,67 @@ DROP FUNCTION IF EXISTS _timescaledb_functions.invalidation_process_hypertable_l
 DROP FUNCTION IF EXISTS _timescaledb_functions.invalidation_process_hypertable_log(integer,integer,regtype,integer[],bigint[],bigint[],text[]);
 DROP FUNCTION IF EXISTS _timescaledb_functions.hypertable_invalidation_log_delete(integer);
 
+--
+-- Rebuild the catalog table `_timescaledb_catalog.continuous_aggs_bucket_function`
+--
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.cagg_get_bucket_function(
+    mat_hypertable_id INTEGER
+) RETURNS regprocedure AS '@MODULE_PATHNAME@', 'ts_continuous_agg_recovery_get_bucket_function' LANGUAGE C STRICT VOLATILE;
+
+-- Since we need now the regclass of the used bucket function, we have to recover it
+-- by parsing the view query by calling 'cagg_get_bucket_function'.
+CREATE TABLE _timescaledb_catalog._tmp_continuous_aggs_bucket_function AS
+    SELECT
+      mat_hypertable_id,
+      _timescaledb_functions.cagg_get_bucket_function(mat_hypertable_id),
+      bucket_width,
+      origin,
+      NULL::text AS bucket_offset,
+      timezone,
+      false AS bucket_fixed_width
+    FROM
+      _timescaledb_catalog.continuous_aggs_bucket_function
+    ORDER BY
+         mat_hypertable_id;
+
+ALTER EXTENSION timescaledb
+    DROP TABLE _timescaledb_catalog.continuous_aggs_bucket_function;
+
+DROP TABLE _timescaledb_catalog.continuous_aggs_bucket_function;
+
+CREATE TABLE _timescaledb_catalog.continuous_aggs_bucket_function (
+  mat_hypertable_id integer NOT NULL,
+  -- The bucket function
+  bucket_func regprocedure NOT NULL,
+  -- `bucket_width` argument of the function, e.g. "1 month"
+  bucket_width text NOT NULL,
+  -- optional `origin` argument of the function provided by the user
+  bucket_origin text,
+  -- optional `offset` argument of the function provided by the user
+  bucket_offset text,
+  -- optional `timezone` argument of the function provided by the user
+  bucket_timezone text,
+  -- fixed or variable sized bucket
+  bucket_fixed_width bool NOT NULL,
+  -- table constraints
+  CONSTRAINT continuous_aggs_bucket_function_pkey PRIMARY KEY (mat_hypertable_id),
+  CONSTRAINT continuous_aggs_bucket_function_mat_hypertable_id_fkey FOREIGN KEY (mat_hypertable_id) REFERENCES _timescaledb_catalog.hypertable (id) ON DELETE CASCADE
+);
+
+INSERT INTO _timescaledb_catalog.continuous_aggs_bucket_function
+  SELECT * FROM _timescaledb_catalog._tmp_continuous_aggs_bucket_function;
+
+DROP TABLE _timescaledb_catalog._tmp_continuous_aggs_bucket_function;
+
+SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.continuous_aggs_bucket_function', '');
+
+GRANT SELECT ON TABLE _timescaledb_catalog.continuous_aggs_bucket_function TO PUBLIC;
+
+ANALYZE _timescaledb_catalog.continuous_aggs_bucket_function;
+
+DROP FUNCTION IF EXISTS _timescaledb_functions.cagg_get_bucket_function(INTEGER);
```

Comment:
Don't we also need an statement like `ALTER EXTENSION timescaledb DROP FUNCTION ...` ??

Reply:
Added